src/write.rs

```diff
@@ -70,16 +74,89 @@ where
     }
 }
 
-impl<T, O> Drop for WriteHandle<T, O>
+/// A **smart pointer** to an owned backing data structure. This makes sure that the
+/// data is dropped correctly (using [`Absorb::drop_second`]).
+///
+/// Additionally it allows for unsafely getting the inner data out using [`into_box()`](Taken::into_box).
+pub struct Taken<T: Absorb<O>, O> {
+    inner: Option<Box<T>>,
+    _marker: PhantomData<O>,
+}
+
+impl<T: Absorb<O> + std::fmt::Debug, O> std::fmt::Debug for Taken<T, O> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Taken")
+            .field(
+                "inner",
+                self.inner
+                    .as_ref()
+                    .expect("inner is only taken in `into_box` which drops self"),
```

Comment:
I'd probably refer to this as "consumes self" rather than "drops self", but it's also fine as-is!